File: TEST data for large repo/repo_Copy-2/Blueprism/BluePrism.Server.Domain.Models/WorkQueueParameters.vb

Imports System.Runtime.Serialization
Imports BluePrism.Server.Domain.Models.DataFilters
Imports BluePrism.Server.Domain.Models.Pagination
Imports Func

<Serializable()>
<DataContract([Namespace]:="bp")>
Public Class WorkQueueParameters
    <DataMember(Name:="s", EmitDefaultValue:=False)>
    Public Property SortBy As WorkQueueSortByProperty
    <DataMember(Name:="o", EmitDefaultValue:=False)>
    Public Property ItemsPerPage As Integer
    <DataMember(Name:="p", EmitDefaultValue:=False)>
    Public Property PageNumber As Integer
    <IgnoreDataMember>
    <CLSCompliantAttribute(False)>
    Public Property NameFilter As DataFilter(Of String)
    <IgnoreDataMember>
    <CLSCompliantAttribute(False)>
    Public Property QueueStatusFilter As DataFilter(Of QueueStatus)
    <IgnoreDataMember>
    <CLSCompliantAttribute(False)>
    Public Property KeyFieldFilter As DataFilter(Of String)
    <IgnoreDataMember>
    <CLSCompliantAttribute(False)>
    Public Property MaxAttemptsFilter As DataFilter(Of Integer)
    <IgnoreDataMember>
    <CLSCompliantAttribute(False)>
   Public Property PendingItemsCountFilter As DataFilter(Of Integer)
    <IgnoreDataMember>
    <CLSCompliantAttribute(False)>
   Public Property LockedItemsCountFilter As DataFilter(Of Integer)
    <IgnoreDataMember>
    <CLSCompliantAttribute(False)>
   Public Property CompletedItemsCountFilter As DataFilter(Of Integer)
    <IgnoreDataMember>
    <CLSCompliantAttribute(False)>
   Public Property  ExceptionedItemsCountFilter As DataFilter(Of Integer)
    <IgnoreDataMember>
    <CLSCompliantAttribute(False)>
    Public Property TotalItemCountFilter As DataFilter(Of Integer)
    <IgnoreDataMember>
    <CLSCompliantAttribute(False)>
   Public Property  AverageItemWorkTimeFilter As DataFilter(Of Integer)
    <IgnoreDataMember>
    <CLSCompliantAttribute(False)>
   Public Property  TotalCaseDurationFilter As DataFilter(Of Integer)
   <IgnoreDataMember>
   Public Property PagingToken As [Option](Of WorkQueuePagingToken)    
End Class

File: TEST data for large repo/repo-Copy5/Blueprism/BluePrism.Api/BluePrism.Api.Domain/WorkQueueParameters.cs
namespace BluePrism.Api.Domain
{
    using Filters;
    using Func;
    using Newtonsoft.Json;
    using PagingTokens;

    public class WorkQueueParameters: IProvideHashCodeForPagingTokenValidation
    {
        public WorkQueueSortByProperty SortBy { get; set; }
        [JsonConverter(typeof(ItemsPerPageConverter))]
        public ItemsPerPage ItemsPerPage { get; set; }
        public Filter<string> NameFilter { get; set; }
        public Filter<QueueStatus> QueueStatusFilter { get; set; }
        public Filter<string> KeyFieldFilter { get; set; }
        public Filter<int> MaxAttemptsFilter { get; set; }
        public Filter<int> PendingItemCountFilter { get; set; }
        public Filter<int> LockedItemCountFilter { get; set; }
        public Filter<int> CompletedItemCountFilter { get; set; }
        public Filter<int> ExceptionedItemCountFilter { get; set; }
        public Filter<int> TotalItemCountFilter { get; set; }
        public Filter<int> AverageWorkTimeFilter {  get; set; }
        public Filter<int> TotalCaseDurationFilter { get; set; }
        [JsonIgnore]
        public Option<PagingToken<int>> PagingToken { get; set; }

        public override int GetHashCode()
        {
            unchecked
            {
                var hashCode = (int)SortBy;
                hashCode = (hashCode * 397) ^ (NameFilter != null ? NameFilter.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (QueueStatusFilter != null ? QueueStatusFilter.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (KeyFieldFilter != null ? KeyFieldFilter.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (MaxAttemptsFilter != null ? MaxAttemptsFilter.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (PendingItemCountFilter != null ? PendingItemCountFilter.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (LockedItemCountFilter != null ? LockedItemCountFilter.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (CompletedItemCountFilter != null ? CompletedItemCountFilter.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (ExceptionedItemCountFilter != null ? ExceptionedItemCountFilter.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (TotalItemCountFilter != null ? TotalItemCountFilter.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (AverageWorkTimeFilter != null ? AverageWorkTimeFilter.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ (TotalCaseDurationFilter != null ? TotalCaseDurationFilter.GetHashCode() : 0);
                hashCode = (hashCode * 397) ^ ItemsPerPage.GetHashCode();
                return hashCode;
            }
        }

        public string GetHashCodeForValidation() => this.ToHmacSha256();
    }
}
